core/embed/unix/drbg.c

```diff
@@ -0,0 +1,68 @@
+/*
```

Comment:
IIRC the only reason why we need the emulator to behave deterministically are the UI tests. In the long run I think we might want to restrict the deterministic behavior to the UI tests only. The emulator should behave randomly in production mode and probably in normal device tests too. I am not saying that that needs to be done as part of this PR, but it's something to think about.

Reply:
Yes, this is something I wasn't sure about so I implemented it the same way it was already in other files. I also didn't want to duplicate the code too much (I also consider creating a directory `core/embed/common` that would contain files that are common for both trezorhal and unix).

Reply:
Having a separate DRBG implementation in trezorhal and unix seems like a pretty clear case of code duplication, but I wasn't sure whether there is possibly some reason for this having to do with deterministic tests as mentioned above. I think there isn't. I'd just take the contents of drbg.c from trezorhal and put it into core/embed/extmod/modtrezorcrypto/rand.c.

Reply:
> but I wasn't sure whether there is possibly some reason for this having to do with deterministic tests as mentioned above. I think there isn't. I'd just take the contents of drbg.c from trezorhal and put it into core/embed/extmod/modtrezorcrypto/rand.c.

I don't understand what you mean. The file `rand.c` is used in both firmware and unix (since 298d90f2cb58ed8c055d416c4d7d688aaab57c89)

Maybe we want to unify both implementations (unix and trezorhal) using `#if FIRMWARE`.

Reply:
We are discussing minimizing code duplication. You were proposing creating `core/embed/common`. I am saying I agree we should get rid of the code duplication and that you can place the common DRBG code in `core/embed/extmod/modtrezorcrypto/rand.c`, because `extmod` plays the role of `common`. The implementations differ only in a few lines of code, which should be distinguished by:

```c
#ifdef TREZOR_EMULATOR
void drbg_set_seed(
...
#else
void drbg_reseed_handler(
...
#endif
```